.github/actions/split_version/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: 'Split Version'
+description: 'Splits the inputed version into major, minor, and patch components'
+inputs:
+  version: 
+    description: 'Version to split'
+    required: true
+  release-branch:
+    description: 'Release branch'
+    required: true
+  existing-version:
+    description: 'Existing version'
+    required: true
+outputs:
+  minor-version:
+    description: "Minor version"
+    value: ${{ steps.split.outputs.minor-version }}
+  patch-version:
+    description: "Patch version"
+    value: ${{ steps.split.outputs.patch-version }}
+  is-pre-release:
+    description: "Is pre-release version"
+    value: ${{ steps.split.outputs.is-pre-release }}
+  release-branch:
+    description: "Is pre-release version"
+    value: ${{ steps.split.outputs.release-branch }}
+  existing-version:
+    description: "Is pre-release version"
+    value: ${{ steps.split.outputs.existing-version }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Split
+      id: split
+      shell: bash
+      run: |
+        git branch
+        major=$(echo ${{ inputs.version }} | cut -d "." -f 1) 
+        minor=$(echo ${{ inputs.version }} | cut -d "." -f 2) 
+        patch=$(echo ${{ inputs.version }} | cut -d "." -f 3 | cut -d "-" -f 1)  
+        minor_version=$major.$minor
+        patch_version=$major.$minor.$patch
```

Comment:
A: Would be helpful to have comments explaining what these vars are given inputs, for example from the old release script:
```
//grab second group (first embedded capture) in first match: 1.4.2-rc.3 -> 1.4.2

//grab second group (first embedded capture) in first match: 1.4.2 -> 1.4
```

Reply:
Added comments to better explain